src/rust/engine/process_execution/src/lib.rs

```diff
@@ -134,6 +134,9 @@ impl AddAssign<UploadSummary> for ExecutionStats {
   }
 }
 
+// TODO(pantsbuild/pants#8039) Need to impl Drop on command runner  so that when the BoxFuture goes out of scope
+// we cancel a potential RPC. So we need to distinguish local vs. remote
```

Comment:
This isn't quite right, sorry. I mixed up what I said in my comment on the review vs in the ticket.

Would recommend putting this TODO on the `remote::CommandRunner`, since it is only relevant there.